src/main/java/com/netflix/frigga/Names.java

```diff
@@ -159,157 +133,74 @@ public Integer getSequence() {
-        if (this == obj)
-            return true;
-        if (obj == null)
-            return false;
-        if (getClass() != obj.getClass())
-            return false;
-        Names other = (Names) obj;
-        if (app == null) {
-            if (other.app != null)
-                return false;
-        } else if (!app.equals(other.app))
-            return false;
-        if (cluster == null) {
-            if (other.cluster != null)
-                return false;
-        } else if (!cluster.equals(other.cluster))
-            return false;
-        if (countries == null) {
-            if (other.countries != null)
-                return false;
-        } else if (!countries.equals(other.countries))
-            return false;
-        if (detail == null) {
-            if (other.detail != null)
-                return false;
-        } else if (!detail.equals(other.detail))
-            return false;
-        if (devPhase == null) {
-            if (other.devPhase != null)
-                return false;
-        } else if (!devPhase.equals(other.devPhase))
-            return false;
-        if (group == null) {
-            if (other.group != null)
-                return false;
-        } else if (!group.equals(other.group))
-            return false;
-        if (hardware == null) {
-            if (other.hardware != null)
-                return false;
-        } else if (!hardware.equals(other.hardware))
-            return false;
-        if (partners == null) {
-            if (other.partners != null)
-                return false;
-        } else if (!partners.equals(other.partners))
-            return false;
-        if (push == null) {
-            if (other.push != null)
-                return false;
-        } else if (!push.equals(other.push))
-            return false;
-        if (redBlackSwap == null) {
-            if (other.redBlackSwap != null)
-                return false;
-        } else if (!redBlackSwap.equals(other.redBlackSwap))
-            return false;
-        if (revision == null) {
-            if (other.revision != null)
-                return false;
-        } else if (!revision.equals(other.revision))
-            return false;
-        if (sequence == null) {
-            if (other.sequence != null)
-                return false;
-        } else if (!sequence.equals(other.sequence))
-            return false;
-        if (stack == null) {
-            if (other.stack != null)
-                return false;
-        } else if (!stack.equals(other.stack))
-            return false;
-        if (usedBy == null) {
-            if (other.usedBy != null)
-                return false;
-        } else if (!usedBy.equals(other.usedBy))
-            return false;
-        if (zone == null) {
-            if (other.zone != null)
-                return false;
-        } else if (!zone.equals(other.zone))
-            return false;
-        return true;
+    public boolean equals(Object o) {
```

Comment:
can safely just consider `group` for equals and hashCode, since that is the original value passed in the constructor, everything else is determined off that value at construction, and group is only non-null if the supplied name was legitimately parseable